apps/challenges/views.py

```diff
@@ -2239,3 +2245,292 @@ def get_challenge_evaluation_cluster_details(request, challenge_pk):
+    except zipfile.BadZipfile:
+        message = "The zip file contents cannot be extracted. Please check the format!"
+        response_data["error"] = message
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    yaml_file_count, yaml_file, extracted_folder_name = get_yaml_files_from_challenge_config(zip_ref)
+
+    if not yaml_file_count:
+        message = "There is no YAML file in zip file you uploaded!"
+        response_data["error"] = message
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if yaml_file_count > 1:
+        message = "There are {0} YAML files instead of one in zip file!".format(yaml_file_count)
+        response_data["error"] = message
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        yaml_file_path = join(base_location, yaml_file)
+        yaml_file_data = read_yaml_file(yaml_file_path, "r")
+    except (yaml.YAMLError, ScannerError) as exc:
+        error_description, line_number, column_number = get_yaml_read_error(exc)
+        message = "\n{} in line {}, column {}\n".format(
+            error_description, line_number, column_number
+        )
+        response_data["error"] = message
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    error_messages = []
+
+    # Check for challenge title.
+    challenge_title = yaml_file_data.get("title")
+    if not challenge_title or len(challenge_title) == 0:
+        message = "Challenge title is missing. Please include."
+        error_messages.append(message)
+
+    # Check for challenge image in yaml file.
+    image = yaml_file_data.get("image")
+    if image and (
+        image.endswith(".jpg")
+        or image.endswith(".jpeg")
+        or image.endswith(".png")
+    ):
+        challenge_image_path = join(
+            base_location, extracted_folder_name, image
+        )
+        if isfile(challenge_image_path):
+            challenge_image_file = read_file_data_as_content_file(challenge_image_path,
+                                                                  "rb", image)
+        else:
+            challenge_image_file = None
+            message = "WARNING: There is no image file."
+            error_messages.append(message)
+    else:
+        challenge_image_file = None
+        message = "ERROR: There is no key for image."
+        error_messages.append(message)
+
+    # check for challenge description file
+    challenge_config_location = join(base_location, extracted_folder_name)
+    is_valid, message = is_challenge_config_yaml_html_field_valid(yaml_file_data,
+                                                                  "description",
+                                                                  challenge_config_location)
+    if not is_valid:
+        error_messages.append(message)
+
+    # check for evaluation details file
```

Comment:
` # check for evaluation details file` -> ` # Check for evaluation details file`